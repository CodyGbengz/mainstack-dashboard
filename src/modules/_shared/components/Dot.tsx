import styled from 'styled-components';

export const Dot = styled.div<{color: string}>`
  background: ${(props) => props.color ?? 'none'};
  border-radius: 0.5rem;
  display: block;
  height: 0.5rem;
  width: 0.5rem;
`;