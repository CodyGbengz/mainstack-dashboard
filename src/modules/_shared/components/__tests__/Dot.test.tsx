import { render } from '@testing-library/react';
import { Dot } from '../Dot';

describe('COMPONENT TEST: Dot', () => {
  const setup = () => render(<Dot color='blue' />);

  test('Should match snapshot', () => {
    const { container } = setup();
    expect(container.parentElement).toMatchSnapshot();
  });
});
