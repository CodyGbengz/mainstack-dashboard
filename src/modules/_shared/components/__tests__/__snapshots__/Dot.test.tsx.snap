// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`COMPONENT TEST: Dot > Should match snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-bdfCkl excHMW"
      color="blue"
    />
  </div>
</body>
`;
